### PR TITLE
Run GPU tests on Windows with Bazel

### PR DESCRIPTION
**Pull request by meteorcloudy**
After this change, we can run:

CPU build + python tests: `tensorflow/tools/ci_build/windows/cpu/pip/build_tf_windows.sh`
CPU C++ tests: `tensorflow/tools/ci_build/windows/cpu/bazel/run_cc_test_windows.sh`
GPU build + python tests: `tensorflow/tools/ci_build/windows/gpu/pip/build_tf_windows.sh`
GPU C++ tests: `tensorflow/tools/ci_build/windows/gpu/bazel/run_cc_test_windows.sh`

### CODE REVIEW

File: tensorflow/tools/ci_build/windows/cpu/pip/build_tf_windows.sh [hunk 1]
```diff
@@ -46,10 +46,9 @@ clean_output_base
 
 run_configure_for_cpu_build
 
-BUILD_OPTS='-c opt --cpu=x64_windows_msvc --host_cpu=x64_windows_msvc --copt=/w --verbose_failures --experimental_ui'
-
 bazel build $BUILD_OPTS tensorflow/tools/pip_package:build_pip_package || exit $?
 
+rm -f ./tensorflow-*.whl
```
Comment (gunan) [inline tensorflow/tools/ci_build/windows/cpu/pip/build_tf_windows.sh]: same comment as in GPU script.
let's build the pip package into a new directory. and avoid rm -f here.

**Review by gunan — CHANGES_REQUESTED** [review]

**Comment by meteorcloudy** [comment]
I refactored all the scripts, please take a look again. :)

File: tensorflow/tools/ci_build/windows/gpu/bazel/run_cc_test_windows.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+# load bazel_test_lib.sh
+source "tensorflow/tools/ci_build/windows/bazel/bazel_test_lib.sh" \
+  || { echo "Failed to source bazel_test_lib.sh" >&2; exit 1; }
+
+clean_output_base
+
+run_configure_for_gpu_build
+
+# Compliling the following test is extremely slow with -c opt
+slow_compiling_test="//tensorflow/core/kernels:eigen_backward_spatial_convolutions_test"
```
Comment (gunan) [inline tensorflow/tools/ci_build/windows/gpu/bazel/run_cc_test_windows.sh]: @mrry @guschmue @vit-stepanovs 
I assume we have the same problem for cmake?

**Review by gunan — APPROVED** [review]

File: tensorflow/tools/ci_build/windows/gpu/bazel/run_cc_test_windows.sh (hunk 2 above)
Comment (vit-stepanovs) [inline tensorflow/tools/ci_build/windows/gpu/bazel/run_cc_test_windows.sh]: Yes, that particular test takes very long to build in CMake too.

**Pull request merged**